__tests__/app.test.js

```diff
@@ -0,0 +1,49 @@
+const seed = require("../db/seeds/seed");
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const data = require("../db/data/test-data/index");
+
+afterAll(() => db.end());
+beforeEach(() => seed(data));
+
+describe("GET /api/topics", () => {
+  test("responds with status 200", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200);
+  });
+  test('responds with status 404 and "Bad path" to an invalid request', () => {
+    return request(app)
+      .get("/api/invalid_request")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad path");
+      });
+  });
+  test("responds with an array of objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .then(({ body }) => {
+        expect(Array.isArray(body)).toBe(true);
+
+        body.forEach((itemObj) => {
+          expect(itemObj).toBeInstanceOf(Object);
+        });
+      });
+  });
+  test('each object has keys: "slug" and "description" and both have string as their value', () => {
+    return request(app)
+      .get("/api/topics")
+      .then(({ body }) => {
+        body.forEach((itemObj) => {
```

Comment:
What happens if the array is empty? We can still iterate over an empty array. When using a forEach in any test you need to assert the length.

Reply:
It's a '/api/topics' endpoint. It should always return an array since it's not a variable like :id?